### PR TITLE
[mono] Automate process for updating SDKs

### PR DESCRIPTION
**Pull request by radical**

### CODE REVIEW

**Comment by radical** [comment]
@akoeplinger @marek-safar Could you please review this too?

**Review by marek-safar — CHANGES_REQUESTED** [review]
Please add documentation how it this supposed to be used and updated

**Comment by radical** [comment]
Added the documentation.

**Review by marek-safar — APPROVED** [review]

File: mono/build/sdks_and_nugets/update_nuget_build_tasks.proj [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+        <RemoveDir Directories="$(TmpPackageDirectory);$(NuGetExtensionsPath);$(NuGetExtensionsPathToolsVersion)" />
+
+        <MakeDir Directories="$(TmpPackageDirectory);$(NuGetExtensionsPath);$(NuGetExtensionsPathToolsVersion)" />
+
+        <Exec
+            Command="nuget install -Source https://dotnet.myget.org/F/cli-deps/api/v3/index.json -Prerelease -OutputDirectory $(TmpPackageDirectory) -Version $(NuGetBuildTasksVersion) NuGet.Build.Tasks"
+            Condition="'$(SkipNuGetBuildTasksInstall)' != 'true'" />
+
+        <Exec Command="find $(TmpPackageDirectory) -type f -iwholename '*net4[567]*' -a ! -name _._ -exec cp {} $(NuGetExtensionsPath) &quot;;&quot;" />
```
Comment (akoeplinger) [inline mono/build/sdks_and_nugets/update_nuget_build_tasks.proj:32]: not sure what the goal here is but the regex won't work for e.g. `net471` or `net48`

File: mono/build/sdks_and_nugets/update_nuget_build_tasks.proj [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        <Exec Command="find $(TmpPackageDirectory) -type f -iwholename '*any/native/*.targets' -exec cp {} $(NuGetExtensionsPath) &quot;;&quot;" />
+
+        <MakeDir Directories="$(NuGetExtensionsPathToolsVersion)/SolutionFile/ImportAfter" />
+        <Exec Command="curl $(CLIBlobBaseURL)/resources/MSBuildImports/15.0/SolutionFile/ImportAfter/Microsoft.NuGet.ImportAfter.targets -o $(NuGetExtensionsPathToolsVersion)/SolutionFile/ImportAfter/Microsoft.NuGet.ImportAfter.targets" />
+
+        <Exec Command="find $(NuGetExtensionsPath) -type f -exec chmod -x {} &quot;;&quot;" />
+        <Exec Command="find $(NuGetExtensionsPathToolsVersion) -type f -exec chmod -x {} &quot;;&quot;" />
```
Comment (akoeplinger) [inline mono/build/sdks_and_nugets/update_nuget_build_tasks.proj:39]: do we really need to chmod _all_ files?

**Review by akoeplinger — COMMENTED** [review]
I think it's ok but I'm a little concerned how we'll ensure we're shipping the right set of files given how much hardcoded logic is in these msbuild targets.

File: mono/build/sdks_and_nugets/update_nuget_build_tasks.proj (hunk 2 above)
Comment (radical) [reply mono/build/sdks_and_nugets/update_nuget_build_tasks.proj:39]: We don't need exec bit on any of the files! And ya, all of them seem to end up with that bit set. This is converted from the bash scripts that I had and will be improved as I use them more. Adding to the repo ensures that they won't be lost and can be maintained better.

**Comment by radical** [comment]
@akoeplinger this is basically a collection of all the steps that I need to take to update the SDKs. And then it is followed by checking `git status` to see what was deleted or look for any new files that might need to be added. And then install+test. So, this isn't really meant as a 100% automated process. It just makes this part of it, which is repetitive, a bit quicker. If things change upstream, like nugets get added/removed, or layouts change then that has to be reflected here. But given that this kinda update needs to be done pretty frequently, the scripts help with reducing the effort. I will make them better as I use them, but you are correct that they are very dependent on what upstream does. And any changes there will have to be tracked here.

The *versions* of the nugets are tracked same as what upstream does, from a single source - the cli repo.

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**